platform/database/tests/database_test.go

```diff
@@ -0,0 +1,281 @@
+		"SELECT * from test_data",
+	)
+
+	if err != nil {
+		t.Fatalf("could not SelectContext: %#v", err)
+	}
+
+	checkTestData := map[string]bool{
+		"first_row":  false,
+		"second_row": false,
+	}
+
+	for _, row := range selectTestData {
+		if row.ID == firstUUID && row.Value == firstValue && checkTestData["first_row"] == false {
+			checkTestData["first_row"] = true
+		} else if row.ID == secondUUID && row.Value == secondValue && checkTestData["second_row"] == false {
+			checkTestData["second_row"] = true
+		} else {
+			t.Fatalf("Expected selectTestData to be first test data and second test data but got %#v", selectTestData)
+		}
+	}
+
+	if !checkTestData["first_row"] {
+		t.Fatalf("expect selectTestData to carry firstValue but got %#v", selectTestData)
+	}
+
+	if !checkTestData["second_row"] {
+		t.Fatalf("expect selectTestData to carry secondValue but got %#v", selectTestData)
+	}
+
+	db.Close()
+
+	// recreating a database each time to prove it can persist across different connections
+	db, err = database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to the database: %#v", err)
+	}
+
+	tx, err := db.Begin()
+
+	if err != nil {
+		t.Fatalf("could not start a transaction: %#v", err)
+	}
+	defer func() { tx.Rollback() }()
+
+	_, err = tx.ExecContext(context.Background(), "UPDATE test_data SET value = $1 WHERE id = $2", "updated_value", firstUUID)
+	if err != nil {
+		t.Fatalf("could not ExecContext in a transaction: %#v", err)
+	}
+
+	// This is inserting a primary key already existing in order to corrupt the transaction
+	_, err = tx.ExecContext(
+		context.Background(),
+		`INSERT INTO test_data (id, value) VALUES ($1, $2);`,
+		firstUUID,
+		firstValue,
+	)
+
+	if err == nil {
+		t.Fatalf("ExecContext should return an error")
+	}
+
+	err = tx.Commit()
+	if err == nil {
+		t.Fatalf("should not be able to commit faulty transaction")
+	}
+
+	err = db.Close()
+	if err != nil {
+		t.Fatalf("COULD NOT CLOSE THE DATABASE AAAAAAAAAAAH: %v", err)
+	}
+
+	db, err = database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to database: %v", err)
+	}
+
+	tx, err = db.Begin()
+
+	err = db.GetContext(
+		context.Background(),
+		&getTestData,
+		`SELECT * from test_data WHERE ID = $1`,
+		firstUUID,
+	)
+	if err != nil {
+		t.Fatalf("could not use getContext %#v", err)
+	}
+
+	if getTestData.Value != firstValue {
+		t.Fatalf("A part of the rollbacked transaction has been committed")
+	}
+
+	if err != nil {
+		t.Fatalf("could not start a transaction: %#v", err)
+	}
+
+	thirdUUID := "5afb4fef-4ecb-4ca3-81fa-10306a9cbb60"
+	thirdValue := "third_value"
+	_, err = db.NamedExecContext(
+		context.Background(),
+		`INSERT INTO test_data (id, value) VALUES (:id, :value)`,
+		testData{ID: thirdUUID, Value: thirdValue},
+	)
+
+	if err != nil {
+		t.Fatalf("could not shoot named exec context to the database: %#v", err)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		t.Fatalf("could not commit transaction")
+	}
+
+	db.Close()
```

Comment:
You are starting a transaction but are using the `db` to execute the statements here